### PR TITLE
Minor updates of `ValidationErrorKind`

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [ ] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).
- [ ] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).
- [ ] A bug fix or other functionality change requiring a patch to `cedar-policy`.
- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)
- [ ] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [ ] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.
- [ ] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)
- [ ] Requires updates, but I do not plan to make them in the near future. (Make sure that your changes are hidden behind a feature flag to mark them as experimental.)
- [ ] I'm not sure how my change impacts `cedar-spec`. (Post your PR anyways, and we'll discuss in the comments.)

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy-validator/src/validation_result.rs [hunk 1]
```diff
@@ -149,18 +149,18 @@ pub enum ValidationErrorKind {
...
     /// head conditions.
     #[error(
-        "Unable to find an applicable action given the policy head constraints{}{}",
+        "unable to find an applicable action given the policy head constraints{}{}",
         if .0.would_in_fix_principal { ". Note: Try replacing `==` with `in` in the principal clause" } else { "" },
         if .0.would_in_fix_resource { ". Note: Try replacing `==` with `in` in the resource clause" } else { "" }
     )]
     InvalidActionApplication(InvalidActionApplication),
-    /// A type error was found by the type checker.
+    /// The type checker found a type error.
```
Comment (andrewmwells-amazon) [inline cedar-policy-validator/src/validation_result.rs:157]: [nit] could probably just say "an error". We already said the type checker found it.

**Review by andrewmwells-amazon — APPROVED** [review]

File: cedar-policy-validator/src/validation_result.rs (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/validation_result.rs]: Good point. Changed.

File: cedar-policy-validator/src/validation_result.rs [hunk 2]
```diff
@@ -125,22 +125,22 @@ impl<'a> SourceLocation<'a> {
...
         .0.actual_entity_type,
         match &.0.suggested_entity_type {
-            Some(s) => format!(", did you mean {}?", s),
+            Some(s) => format!(", did you mean `{}`?", s),
             None => "".to_string()
         }
     )]
     UnrecognizedEntityType(UnrecognizedEntityType),
-    /// An action id was seen in a policy but was not found in the schema.
+    /// A policy contains an action id that is not declared in the schema.
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/validation_result.rs:138]: Is the phrase "action id" consistent with our other error messages? Searching for "entity id" doesn't find many results, so maybe just "action" or "action entity" would be better.

**Review by john-h-kastner-aws — APPROVED** [review]
One question on consistency of our phrasing, but otherwise looks good.

**Review by cdisselkoen — APPROVED** [review]

File: cedar-policy-validator/src/validation_result.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/validation_result.rs]: Good catch. Fixed.

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**